core/src/shred_fetch_stage.rs

```diff
@@ -373,6 +394,22 @@ pub(crate) fn receive_repair_quic_packets(
     }
 }
 
+#[must_use]
+fn should_drop_legacy_shreds(
+    shred_slot: Slot,
+    feature_set: &FeatureSet,
+    epoch_schedule: &EpochSchedule,
+) -> bool {
+    match feature_set.activated_slot(&feature_set::drop_legacy_shreds::id()) {
+        None => false,
+        Some(feature_slot) => {
+            let feature_epoch = epoch_schedule.get_epoch(feature_slot);
+            let shred_epoch = epoch_schedule.get_epoch(shred_slot);
+            feature_epoch < shred_epoch
```

Comment:
one epoch delay is intentional, correct?

Reply:
yes, at epoch boundary, if shreds are from the next epoch, we don't know if the feature is activated or not,